fst_lookup/fst.py

```diff
@@ -210,35 +213,46 @@ def __init__(
         self.arcs_from = arcs_from
 
     def __iter__(self) -> Iterator[RawTransduction]:
-        yield from self._accept(self.initial_state, [], [{}])
+        yield from self._accept(self.initial_state, [], [{}], [])
 
     def _accept(
         self,
         state: StateID,
         transduction: List[Symbol],
         flag_stack: List[Dict[str, str]],
+        stems: List[Symbol]
     ) -> Iterable[RawTransduction]:
         # TODO: Handle a maximum transduction depth, for cyclic FSTs.
         if state in self.accepting_states:
             if len(self.symbols) > 0:
                 return
-            yield tuple(transduction)
+            combined_stems = ''.join([str(x) for x in stems])
+            combined_stems = re.sub(r"\++", "+", combined_stems)
+            if combined_stems.endswith("+"):
+                combined_stems = combined_stems[:-1]
+            if combined_stems.startswith("+"):
+                combined_stems = combined_stems[1:]
+            yield transduction, combined_stems
 
         for arc in self.arcs_from[state]:
             input_label = self.get_input_label(arc)
 
             if input_label is Epsilon:
                 # Transduce WITHOUT consuming input
                 transduction.append(self.get_output_label(arc))
-                yield from self._accept(arc.destination, transduction, flag_stack)
+                stems.append('+')
```

Comment:
Overall, this change makes some VERY strong assumptions about the FST. This repo already makes strong assumptions about the FST, but the last bug fix was to relax some of those assumptions (!!!), so I am bit hesitant to accept these changes as they are now.

Reply:
Would you please give me more information about these assumptions? Does this change make another assumption or does it depend on some assumption that may change in the near future?